### PR TITLE
final rx java removal

### PR DESCRIPTION
**Pull request by OwenLindsell**

### CODE REVIEW

**Comment by OwenLindsell** [comment]
> Hi @OwenLindsell . Thanks for this! Can you check the logging output from CI run please? The CI run started erring (not failing) due to excessive logging.

Which errors specifically? There are a lot of errors in the log 😄 
You think the increase in logging is caused by this PR?

**Comment by mikkokar** [comment]
> > Hi @OwenLindsell . Thanks for this! Can you check the logging output from CI run please? The CI run started erring (not failing) due to excessive logging.
> 
> Which errors specifically? There are a lot of errors in the log 😄
> You think the increase in logging is caused by this PR?

Master build 2122:

```
B/s)
[INFO] Downloading from maven-central-mirror: https://maven-central.storage-download.googleapis.com/maven2/org/apache/maven/doxia/doxia-module-xhtml/1.4/doxia-module-xhtml-1.4.pom
[INFO] Downloaded from maven-central-mirror: https://maven-central.storage-download.googleapis.com/maven2/org/apache/maven/doxia/doxia-module-xhtml/1.4/doxia-module-xhtml-1.4.pom (0 B at 0 B/s)
[INFO] Downloading from maven-central-mirror: https://maven-central.storage-download.googleapis.com/maven2/org/apache/maven/doxia/doxia-modules/1.4/doxia-modules-1.4.pom
[INFO] Downloaded from maven-central-mirror: https://maven-central.storage-download.googleapis.com/maven2/org/apache/maven/doxia/doxia-modules/1.4/doxia-modules-1.4.pom (0 B at 0 B/s)
[INFO] Downloading from maven-central-mirror: https://maven-central.storage-download.googleapis.com/maven2/org/apache/maven/doxia/doxia-module-fml/1

The job exceeded the maximum log length, and has been terminated.
```

**Comment by OwenLindsell** [comment]
Ah ok, this is not actually related to this PR, but I see it's a blocker for merging any PR.  We have a story for this in Trello:
https://trello.com/c/mnChSQv1/150-excessive-logging-in-ci-environment

Can I get an approval for this PR and we will only merge once the above story is complete.

File: components/server/src/main/java/com/hotels/styx/server/netty/connectors/HttpResponseWriter.java [hunk 1]
```diff
@@ -157,7 +156,12 @@ private void onWriteEmptyLastChunkOutcome(ChannelFuture writeOp) {
             return future;
         } catch (Throwable cause) {
             LOGGER.warn("Failed to convert response headers. response={}, Cause={}", new Object[]{response, cause});
-            toObservable(response.body()).forEach(it -> Buffers.toByteBuf(it).release());
+            response.body().subscribe(new BaseSubscriber<Buffer>() {
+                @Override
+                protected void hookOnNext(Buffer b) {
+                    Buffers.toByteBuf(b).release();
+                }
+            });
```
Comment (mikkokar) [inline components/server/src/main/java/com/hotels/styx/server/netty/connectors/HttpResponseWriter.java:164]: I think we should use the `ByteStream.drop` instead:

```
            Flux.from(response.body()
                    .drop())
                    .subscribe();
```

Comment (OwenLindsell) [reply components/server/src/main/java/com/hotels/styx/server/netty/connectors/HttpResponseWriter.java]: Yes, much neater! done :)

**Pull request merged**